tests/RobotFramework/tests/cumulocity/mapper_recovery/recover_and_publish_software_update_message.robot

```diff
@@ -0,0 +1,37 @@
+*** Settings ***
+Resource    ../../../resources/common.resource
+Library    Cumulocity
+Library    ThinEdgeIO
+
+Test Tags    theme:c8y    theme:mapper recovery 
+Suite Setup    Custom Setup
+Test Teardown    Get Logs
+
+*** Test Cases ***
+Mapper recovers and processes output of ongoing software update request
+    [Documentation]    C8y-Mapper receives a software update request, 
+    ...                Delegates operation to tedge-agent and gets back `executing` status
+    ...                And then goes down (here purposefully stopped).
+    ...                Mean while the agent processes the update message and publishes the software update message
+    ...                After some time mapper recovers and pushes the result to c8y cloud
+    ...                Verify that the rolldice package is installed or not
+    ${timestamp}=        Get Unix Timestamp
+    ThinEdgeIO.Service Should Be Running    tedge-mapper-c8y    
+    ${OPERATION}=    Install Software        rolldice
+    Should Have MQTT Messages    tedge/commands/res/software/update    message_contains=executing    date_from=${timestamp}
+    ThinEdgeIO.Stop Service    tedge-mapper-c8y    
+    Should Have MQTT Messages    tedge/commands/res/software/update    message_contains=successful    date_from=${timestamp}
+    ThinEdgeIO.Start Service    tedge-mapper-c8y
+    ThinEdgeIO.Service Should Be Running    tedge-mapper-c8y
+    Operation Should Be SUCCESSFUL           ${OPERATION}    timeout=60
+    Device Should Have Installed Software    rolldice
+ 
+*** Keywords ***
+
+Custom Setup
+    ${DEVICE_SN}=    Setup
+    Set Suite Variable    $DEVICE_SN 
+    Device Should Exist                      ${DEVICE_SN}
+    Service Health Status Should Be Up    tedge-mapper-c8y
+    # This acts as a custom apt plugin
+    ThinEdgeIO.Transfer To Device     ${CURDIR}/custom_sw_plugin.sh    /etc/tedge/sm-plugins/apt
```

Comment:
Ideally we should use another sm-plugin name to make it more obvious that it is a fake plugin.

```suggestion
    ThinEdgeIO.Transfer To Device     ${CURDIR}/custom_sw_plugin.sh    /etc/tedge/sm-plugins/dummy
```

Though if you do change it, then you will also need to change the `Install Software` line to make use of the custom prefix

```robot
${OPERATION}=    Install Software    rolldice,1.0.0::dummy
```

Reply:
Make sense, I have updated the test.